### ISSUE TITLE
NullReferenceException in RedundantCheckBeforeAssignmentAnalyzer

### ISSUE THREAD

**Issue by neoGeneva** [icsharpcode/RefactoringEssentials#135]
I'm getting a null reference exception, and it looks like it's part of RECS0134

I'm not 100% sure what's causing this issue, but let me know if you need any more info from me.

Stack Trace:
```
System.NullReferenceException: Object reference not set to an instance of an object.
   at RefactoringEssentials.CSharp.Diagnostics.RedundantCheckBeforeAssignmentAnalyzer.CheckConditionAndAssignment(SyntaxNodeAnalysisContext nodeContext, AssignmentExpressionSyntax assignment, BinaryExpressionSyntax condition)
   at RefactoringEssentials.CSharp.Diagnostics.RedundantCheckBeforeAssignmentAnalyzer.TryGetDiagnostic(SyntaxNodeAnalysisContext nodeContext, Diagnostic& diagnostic)
   at RefactoringEssentials.CSharp.Diagnostics.RedundantCheckBeforeAssignmentAnalyzer.<>c.<Initialize>b__3_0(SyntaxNodeAnalysisContext nodeContext)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c__DisplayClass35_0`1.<ExecuteSyntaxNodeAction>b__1()
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock(DiagnosticAnalyzer analyzer, Action analyze)
```

**Comment by Rpinski** [comment]
I've fixed some exceptions found in RCS0134 (70b1cc40a3a48c682be8402dac457e153e91f216), but without any code example I can't say if fixes your exception, too. Maybe you can try latest development build:
http://vsixgallery.com/extension/NR6Pack.Refactorings.Vsix..cd27e59b-5c56-4238-8931-c9bb745a7a7f/

**Comment by neoGeneva** [comment]
Hey, I cloned the repo a v2.1 and can confirm that your commit does fix my issue, specifically the change on line 127 of RedundantCheckBeforeAssignmentAnalyzer.cs, thanks!
